modules/scala-http4s/src/main/scala/dev/guardrail/generators/scala/http4s/Http4sServerGenerator.scala

```diff
@@ -608,7 +762,62 @@ class Http4sServerGenerator private (version: Http4sVersion)(implicit Cl: Collec
       val responseType = ServerRawResponse(operation)
         .filter(_ == true)
         .fold[Type](t"${Term.Name(resourceName)}.$responseCompanionType")(Function.const(t"Response[F]"))
-      val orderedParameters: List[List[LanguageParameter[ScalaLanguage]]] = List((pathArgs ++ qsArgs ++ bodyArgs ++ formArgs ++ headerArgs).toList) ++
+      val authContext: Option[(LanguageParameter[ScalaLanguage], Term => Term)] = authImplementation match {
+        case Disable => None
+        case Native  => Some((LanguageParameter.fromParam(param"authContext: $authContextTypeName"), identity _))
+        case Custom =>
+          securityRequirements.map({ sr =>
+            val arg                           = LanguageParameter.fromParam(param"authContext: $authContextTypeName")
+            val securityRequirements          = renderCustomSecurityRequirements(sr)
+            val authContextParam              = param"${arg.paramName}"
+            val authTransformer: Term => Term = inner => q"""
+                authenticationMiddleware($securityRequirements, ${sr.optional}, req).flatMap { $authContextParam =>
```

Comment:
This is the last sticking point for me. `${sr.optional}` being passed in as a `Boolean` into `authenticationMiddleware` is really confusing, as a user. If authentication is optional, shouldn't there just be a `recover` on whatever `$authContextTypeName` is returned? `Some(Unauthorized)` => `None`?

Reply:
I took a stab at resolving this. There's something nagging at me about having both `SecurityExposure` and `AuthImplementation` passed in as separate parameters to a bunch of parameters. I've not changed any of the underlying logic, though reducing the number of `Boolean`'s exposed to users is better. I'll come back to this soon (I'd like to get this merged and on its way), but offering here in case anyone is interested in taking a look.

Reply:
Haven't found a great way to resolve this, so I'm going to merge as is.